### PR TITLE
fix(aoc): Update unread message count when messages are deleted

### PR DESCRIPTION
**Pull request by Eearslya**
## Summary of Changes

This fixes an issue where the `COMPANY MSG` memo will not clear if an AOC message is deleted before it is read.

## Additional context

Discord username (if different from GitHub): Eearslya#7831

## Testing instructions

1. Use the AOC menu to request a METAR or ATIS report.
1. Confirm `COMPANY MSG` appears on the ECAM after a few moments.
1. Use the AOC menu to open and read the message. Confirm `COMPANY MSG` disappears from ECAM.
1. Repeat steps 1-2, then use the CLR key to delete the message without reading it. Confirm `COMPANY MSG` disappears from ECAM.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/systems/atsu/src/AOC.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ export class Aoc {
     public removeMessage(uid: number): boolean {
         const index = this.messageQueue.findIndex((element) => element.UniqueMessageID === uid);
         if (index !== -1) {
+            if (this.messageQueue[index].Confirmed === false) {
+                const cMsgCnt = SimVar.GetSimVarValue('L:A32NX_COMPANY_MSG_COUNT', 'Number');
+                SimVar.SetSimVarValue('L:A32NX_COMPANY_MSG_COUNT', 'Number', cMsgCnt <= 1 ? 0 : cMsgCnt - 1);
+            }
```
Comment (tracernz) [inline fbw-a32nx/src/systems/atsu/src/AOC.ts:39]: I suggest a less error-prone patch, add a new method called updateMessageCount or something and call it in each of the functions that mutates the queue (better would be to wrap the queue in a nice abstraction but that's an even bigger patch). Probably also call it in an init function that's called (indirectly) from onConnectedCallback in the instrument so it clears the count when the instrument is reloaded etc.
```ts
private async updateMessageCount(): Promise<unknown> {
    const count = this.messageQueue.reduce((c, m) => m.Confirmed ? c : c + 1, 0);
    return SimVar.SetSimVarValue('L:A32NX_COMPANY_MSG_COUNT', 'number', count);
}
```

Comment (svengcz) [reply fbw-a32nx/src/systems/atsu/src/AOC.ts:39]: We can go for the easiest fix.
Will be replaced with the a new logic in next ATSU update in my PR

Comment (Eearslya) [reply fbw-a32nx/src/systems/atsu/src/AOC.ts]: Fair enough, good suggestion. The code duplication did cross my mind, but I wasn't sure if it was worth a new function.

**Review by svengcz — APPROVED** [review]
Lgtm

**Comment by brewers10** [comment]
Quality Assurance Trainee Report

Discord          : brewers_10#5706
Object of testing: #7783 
Tier of Testing  : 1
Date             : 11/02/2023

Testing Process: 
Loaded into KFBW. Requested METAR report for KLAX. COMPANY MSG appeared on the ECAM. Read the METAR. COMPANY MSG disappeared. Repeated steps again: requested METAR for KLAX. COMPANY MSG appeared on ECAM. Cleared the message without reading it. COMPANY MSG disappeared.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Pull request merged**